### PR TITLE
Fix Hapi request types causing ESLint warnings etc

### PR DESCRIPTION
**Pull request by colinrotherham**
Lots of commits I'd recommend going through one by one. Stacked on top of:

* https://github.com/DEFRA/forms-runner/pull/471

The main addition is for `FormRequest` (GET) and `FormRequestPayload` (POST) types for Hapi

```patch
  private summaryDetails(
-   request: Request,
+   request: FormRequest | FormRequestPayload,
    model: FormModel,
    state: FormSubmissionState,
    relevantPages: PageControllerClass[]
  ) {
```

Or to fix a few POST routes accidentally using GET options

```patch
      path: '/{slug}/{path}/summary',
      handler: postListSummaryHandler,
      options: {
-       ...getRouteOptions,
+       ...postRouteOptions,
        validate: {
          params: Joi.object().keys({
            slug: slugSchema,
```

## Before
Output from `npm run lint:js`

```console
✖ 115 problems (7 errors, 108 warnings)
```

## After
Output from `npm run lint:js`

```console
✖ 100 problems (7 errors, 93 warnings)
```

### CODE REVIEW

File: src/server/plugins/engine/components/CheckboxesField.ts [hunk 1]
```diff
@@ -43,43 +43,42 @@ export class CheckboxesField extends SelectionControlField {
...
     let { items, value } = viewModel
 
-    let payloadItems: string[] | number[] = []
-
     // Allow strings (temporarily) via controller renderWithErrors()
-    payloadItems = Array.isArray(value) ? value : payloadItems
-    payloadItems = typeof value === 'string' ? [value] : payloadItems
+    const payloadItems = [value]
+      .flat()
+      .filter((value) => typeof value === 'string' || typeof value === 'number')
```
Comment (colinrotherham) [inline src/server/plugins/engine/components/CheckboxesField.ts:75]: We can tweak this bit should we keep https://github.com/DEFRA/forms-runner/pull/296

By sanitising the payload we'll only have array values

**Review by davidjamesstone — APPROVED** [review]

File: src/server/plugins/engine/models/FormModel.ts [hunk 2]
```diff
@@ -161,22 +146,8 @@ export class FormModel {
    * instantiates a Page based on {@link Page}
    */
   makePage(pageDef: Page): PageControllerClass {
-    if (pageDef.controller) {
-      const PageController = getPageController(pageDef.controller)
-
-      if (!PageController) {
```
[the remark this answers is not in the document]
Comment (colinrotherham) [reply src/server/plugins/engine/models/FormModel.ts:167]: Ah glad you spotted this too. We're safe to remove it 🎉 

We added 2x new helpers to `@defra/forms-model` recently:

```js
isControllerName()       // Checks if the path or class name is known
controllerNameFromPath() // Returns class names known paths to supported class names
```

They're all type checked against [**model/src/pages/controller-types.ts**](https://github.com/DEFRA/forms-designer/blob/c7dd495cd4f9eb8cefd95d175648527100cb6d97/model/src/pages/controller-types.ts)

We run [`getPageController()`](https://github.com/DEFRA/forms-runner/blob/29222ccb7d1a6254e311f200cb8f01c087adcd8e/src/server/plugins/engine/pageControllers/helpers.ts#L23-L33) → [`controllerNameFromPath()`](https://github.com/DEFRA/forms-designer/blob/e578f15593852b972b043e409f8d4d9356ae873a/model/src/pages/helpers.ts#L110-L117) → [`isPageController`](https://github.com/DEFRA/forms-runner/blob/29222ccb7d1a6254e311f200cb8f01c087adcd8e/src/server/plugins/engine/pageControllers/helpers.ts#L10-L14)

Which ultimately falls back to the default `PageController` when all else fails:

```js
if (!isPageController(controllerName)) {
  return PageControllers.PageController
}
```

File: src/server/plugins/engine/models/SummaryViewModel.ts [hunk 3]
```diff
@@ -124,7 +127,7 @@ export class SummaryViewModel {
           addRepeaterItem(page, request, model, state, items)
         } else {
           for (const component of page.components.formItems) {
-            const item = Item(request, component, state, page, model)
+            const item = Item(component, state, page, model)
```
[the remark this answers is not in the document]
Comment (colinrotherham) [reply src/server/plugins/engine/models/SummaryViewModel.ts:130]: Great idea. I'd love to bring repeater items back into the same constructor too

Whilst they're don't have a `component` they still share the usual name/title/label fields

Hopefully nice and easy to bring them back into one. Can we save it for a wider chat?

File: src/server/plugins/engine/models/FormModel.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (colinrotherham) [reply src/server/plugins/engine/models/FormModel.ts:167]: Would be good to chat about migration (and whether any JSON still uses legacy paths) soon

Got this PR ready for when the time comes:

* https://github.com/DEFRA/forms-runner/pull/333

**Pull request merged**